bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,250 @@
+
+        1) use the command \".lovefest sub\" to get the lovefest role.
+        2) use the command \".lovefest unsub\" to get rid of the lovefest role.
+            """
+        await ctx.invoke(self.bot.get_command("help"), "lovefest")
+
+    @lovefest_role.command(name="sub")
+    async def add_role(self, ctx):
+        """
+        This command adds the lovefest role.
+        """
+        user = ctx.author
+        role = discord.utils.get(ctx.guild.roles, id=Lovefest.role_id)
+        if Lovefest.role_id not in [role.id for role in ctx.message.author.roles]:
+            await user.add_roles(role)
+            await ctx.send("The Lovefest role has been added !")
+        else:
+            await ctx.send("You already have the role !")
+
+    @lovefest_role.command(name="unsub")
+    async def remove_role(self, ctx):
+        """
+        This command removes the lovefest role.
+        """
+        user = ctx.author
+        role = discord.utils.get(ctx.guild.roles, id=Lovefest.role_id)
+        if Lovefest.role_id not in [role.id for role in ctx.message.author.roles]:
+            await ctx.send("You dont have the lovefest role.")
+        else:
+            await user.remove_roles(role)
+            await ctx.send("The lovefest role has been successfully removed !")
+
+    @commands.cooldown(1, 1800, BucketType.user)
+    @commands.group(name='bemyvalentine', invoke_without_command=True)
+    async def send_valentine(self, ctx, user: typing.Optional[discord.Member] = None, *, valentine_type=None):
+        """
+        This command sends valentine to user if specified or a random user having lovefest role.
+
+        syntax: .bemyvalentine [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        (optional)
+
+        example: .bemyvalentine (sends valentine as a poem or a compliment to a random user)
+        example: .bemyvalentine Iceman#6508 p (sends a poem to Iceman)
+        example: .bemyvalentine Iceman Hey I love you, wanna hang around ? (sends the custom message to Iceman)
+        NOTE : AVOID TAGGING THE USER MOST OF THE TIMES.JUST TRIM THE '@' when using this command.
+        """
+        if ctx.guild is None:
+            # This command should only be used in the server
+            msg = "You are supposed to use this command in the server."
+            return await ctx.send(msg)
+
+        if user:
+            if Lovefest.role_id not in [role.id for role in user.roles]:
+                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+                return await ctx.send(message)
+
+        if user == ctx.author:
+            # Well a user cant valentine himself/herself.
+            return await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+
+        emoji_1, emoji_2 = self.random_emoji()
+        Role = discord.utils.get(ctx.guild.roles, id=Lovefest.role_id)
```

Comment:
This needs a different, more specific name. `Role` would be assumed to reference the abstract `discord.Role` object rather than the object that represents the lovefest role.